### PR TITLE
feature(testing): Use candidate versions of pepr in E2E tests

### PR DESCRIPTION
**Pull request by samayer12**

### CODE REVIEW

File: _helpers/dev/cli.mts [hunk 1]
```diff
@@ -43,6 +43,20 @@ const test = program.command('test')
       'args to pass to test runner (e.g. --passthru=\'--testNamePattern="testName()"\')'
     )
   )
+  .hook('preAction', () =>{
+    const installPath = execSync('pwd').toString().trim().concat('/..') // Top-level package.json
+    execSync('npm install', {cwd: installPath})
+    if(!process.env.CI)
+    {
+      //Local setup, since we're not in CI
+      //TODO: Hardcoded paths need to be generalized
```
Comment (samayer12) [inline _helpers/dev/cli.mts:52]: Pathing issues

File: use-arbitrary-pepr-version/capabilities/versions.e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      expect(result).toContain('0.36.0');
+    })
+    it('shows the help menu without --unpublished', () =>{
+      const result = execSync('npx pepr init --help').toString()
+      expect(result).not.toContain('--unpublished')
+    })
+  })
+  describe('when pepr is a local dev copy', () => { 
+    const peprAlias = "file:../pepr-0.0.0-development.tgz"
+    const installPath = execSync('pwd').toString().trim().concat('/..') // Top-level package.json
```
Comment (samayer12) [inline use-arbitrary-pepr-version/capabilities/versions.e2e.test.ts:51]: Pathing issues

File: use-arbitrary-pepr-version/capabilities/versions.e2e.test.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    beforeAll(() =>{
+      if(process.env.CI){
+        console.log("Testing In CI with .tgz!")
+      }
+      else{
+        console.log("Local testing with .tgz!")
```
Comment (samayer12) [inline use-arbitrary-pepr-version/capabilities/versions.e2e.test.ts:57]: log statements to remove

File: use-arbitrary-pepr-version/capabilities/versions.e2e.test.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      }
+      expect(execSync(`ls -l ../`, {cwd: installPath}).toString()).toContain('pepr-0.0.0-development.tgz')
```
Comment (samayer12) [inline use-arbitrary-pepr-version/capabilities/versions.e2e.test.ts:59]: pathing issue

File: _helpers/dev/cli.mts [hunk 5]
```diff
@@ -31,6 +32,18 @@ const test = program.command('test')
       .choices(['unit', 'e2e', 'all'])
       .default('all')
   )
+  .addOption(
+    new Option(
```
Comment (samayer12) [inline _helpers/dev/cli.mts:36]: Enforce mutex

File: _helpers/dev/cli.mts [hunk 6]
```diff
@@ -43,6 +56,32 @@ const test = program.command('test')
       'args to pass to test runner (e.g. --passthru=\'--testNamePattern="testName()"\')'
     )
   )
+  .hook('preAction', (thisCommand) =>{
+    const peprExcellentExamplesRepo = findUpSync('pepr-excellent-examples', {type: 'directory'})
+    execSync('npm install', {cwd: peprExcellentExamplesRepo})
+    if(!process.env.CI)
```
Comment (samayer12) [inline _helpers/dev/cli.mts:62]: replace with use of custom package instead of envar check

File: _helpers/dev/cli.mts [hunk 7]
```diff
@@ -43,6 +56,32 @@ const test = program.command('test')
...
+      if(thisCommand.opts().localPackage){
+        const peprRepoLocation = findUpSync('pepr', {type: 'directory'})
+        const peprBuild = 'pepr-0.0.0-development.tgz'
+        const peprContainerImage = 'pepr:dev'
+
+        execSync('npm run build > /dev/null 2>&1', {cwd: peprRepoLocation})
+        execSync(`cp ${peprRepoLocation}/${peprBuild} ${peprExcellentExamplesRepo}`)
+        console.log(`Pepr Build under test: ${execSync(`shasum ${peprExcellentExamplesRepo}/${peprBuild}`).toString()}\n` +
+        `Pepr Image under test: ${execSync(`docker inspect --format=\'{{.Id}}\' ${peprContainerImage}`).toString()}`)
+        process.env.LOCAL_PACKAGE = `${peprExcellentExamplesRepo}/${peprBuild}`
```
Comment (samayer12) [inline _helpers/dev/cli.mts:81]: assign to just one envar? The only difference is if we build or use a provided file.

File: test-specific-version/README.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+# Use Arbitrary Pepr Version
+
+This example shows how to run tests across multiple versions of pepr, to include release candidates on a branch not yet merged into `main`.
```
Comment (samayer12) [inline test-specific-version/README.md:3]: Update docs

**Review by cmwylie19 — APPROVED** [review]
lgtm

**Comment by cmwylie19** [comment]
syncro review

**Pull request merged**